Make path to includes relative (honor CMAKE_INSTALL_PREFIX)

cfsmp3 opened the pull request:
Install paths must be relative, otherwise CMAKE_INSTALL_PREFIX is ignored and

```
make install
```

tries to write to /usr/include regardless CMAKE_INSTALL_PREFIX. 

```
 TEST="0" cmake  ./  -DCMAKE_INSTALL_PREFIX=/tmp/hello && make && make install
-- Configuring done
-- Generating done
-- Build files have been written to: /home/carlos/deleteme/prometheus-client-c-0.1.1/prom
[100%] Built target prom
[100%] Built target prom
Install the project...
-- Install configuration: ""
-- Up-to-date: /tmp/hello/lib/libprom.so
-- Up-to-date: /usr/include/
CMake Error at cmake_install.cmake:68 (file):
  file INSTALL cannot set permissions on "/usr/include/": Operation not
  permitted.

make: *** [Makefile:97: install] Error 1
```

Also removed trailing slash (to prevent // in paths).

miroswan commented:
This change appears to have broken the build:

https://travis-ci.org/github/digitalocean/prometheus-client-c/jobs/699530011

cfsmp3 commented:
> This change appears to have broken the build:

Thanks. Fixed.
https://travis-ci.org/github/digitalocean/prometheus-client-c/builds/699880303

jtojnar commented on prom/CMakeLists.txt:
```diff
@@ -137,4 +137,4 @@ set(CPACK_PACKAGE_HOMEPAGE_URL https://github.internal.digitalocean.com/timeseri
 include(CPack)
 install(TARGETS prom ARCHIVE)
-install(DIRECTORY include/ DESTINATION /usr/include/)
+install(DIRECTORY include/ DESTINATION include)
```
It is still better to use [`GNUInstallDirs`](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) module so that distros can choose the installation location (e.g. using multiple outputs in Nixpkgs).

miroswan commented:
@jtojnar and @cfsmp3 What if we added the following? 

```
include(GNUInstallDirs)
install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

I believe this would give you the most flexibility whilst sticking to convention.

cfsmp3 commented:
> @jtojnar and @cfsmp3 What if we added the following?

> include(GNUInstallDirs)
> install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
> ```
> I believe this would give you the most flexibility whilst sticking to convention.

Works. I've updated the PR.

MattIPv4 commented:
Hey, @cfsmp3 - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉
